doc/shared/sunlinsol/SUNLinSol_Ginkgo.rst

```diff
@@ -0,0 +1,192 @@
+
+   sundials::ginkgo::LinearSolver<GkoSolverType, GkoMatrixType> LS{
+      gko_solver_factory, sunctx};
+
+Finally, we can pass the instance of :cpp:type:`sundials::ginkgo::LinearSolver` to any function
+expecting a ``SUNLinearSolver`` object through the implicit conversion operator or explicit conversion function.
+
+.. note::
+
+  The ``N_Vector`` to use with the SUNLINEARSOLVER_GINKGO module depends on the ``gko::Executor`` utilized.
+  I.e., when using the ``gko::CudaExecutor`` you should use a CUDA capable N_Vector  (e.g., :numref:`NVectors.CUDA`),
+  when using  ``gko::OmpExecutor`` you should use a CPU based N_Vector, etc.
+
+.. code-block:: cpp
+
+   // Attach linear solver and matrix to CVODE.
+   //
+   // Implicit conversion from sundials::ginkgo::LinearSolver<GkoSolverType, GkoMatrixType>
+   // to a SUNLinearSolver object is done.
+   //
+   // For details about creating A see the SUNMATRIX_GINKGO module. 
+   CVodeSetLinearSolver(cvode_mem, LS, A);
+
+   // Alternatively with explicit conversion of LS to a SUNLinearSolver
+   // and A to a SUNMatrix:
+   CVodeSetLinearSolver(cvode_mem, LS->Convert(), A->Convert());
+
+.. warning::
+
+  :c:func:`SUNLinSolFree` should never be called on a ``SUNLinearSolver`` that was created via conversion
+  from a ``sundials::ginkgo::LinearSolver``. Doing so may result in a double free. 
+
+
+.. _SUNLinSol.Ginkgo.API:
+
+SUNLINEARSOLVER_GINKGO API
+--------------------------
+
+In this section we list the public API of the :cpp:type:`sundials::ginkgo::LinearSolver` class. 
+
+.. cpp:class:: template<class GkoSolverType, class GkoMatrixType> \
+               LinearSolver : public ConvertibleTo<SUNLinearSolver>
+
+   .. cpp:function:: LinearSolver() = default;
+
+      Default constructor - means the solver must be moved to
+
+   .. cpp:function:: LinearSolver(std::shared_ptr<typename GkoSolverType::Factory> gko_solver_factory, SUNContext sunctx)
+
+      Constructs a new LinearSolver from a Ginkgo solver factory
+
+      :param gko_solver_factory: The Ginkgo solver factory (typically `gko::matrix::<type>::Factory``)
+      :param sunctx: The SUNDIALS simulation context (:c:type:`SUNContext`)
+
+   .. cpp:function:: LinearSolver(LinearSolver&& that_solver) noexcept
+
+      Move constructor
+
+   .. cpp:function:: LinearSolver& operator=(LinearSolver&& rhs)
+
+      Move assignment
+
+   .. cpp:function:: ~LinearSolver() override = default
+
+      Default destructor
+
+   .. cpp:function:: operator SUNLinearSolver() override
+
+      Implicit conversion to a :c:type:`SUNLinearSolver`
+   
+   .. cpp:function:: operator SUNLinearSolver() const override
+
+      Implicit conversion to a :c:type:`SUNLinearSolver`
+
+   .. cpp:function:: SUNLinearSolver Convert() override
+
+      Explicit conversion to a :c:type:`SUNLinearSolver`
+
+   .. cpp:function:: SUNLinearSolver Convert() const override
+
+      Explicit conversion to a :c:type:`SUNLinearSolver`
```

Comment:
The explanation text is identical because they do identical things. Which one you use is just a matter of C++ semantics.  See https://github.com/LLNL/sundials/pull/205/files#r984982868.